src/coreclr/vm/dacenumerablehash.inl

```diff
@@ -367,11 +367,9 @@ DPTR(VALUE) DacEnumerableHashTable<DAC_ENUM_HASH_ARGS>::BaseFindNextEntryByHash(
     PTR_VolatileEntry pVolatileEntry = dac_cast<PTR_VolatileEntry>(pContext->m_pEntry);
     iHash = pVolatileEntry->m_iHashValue;
 
-    // Iterate over the bucket chain.
-    while (pVolatileEntry->m_pNextEntry)
+    // Iterate over the rest ot the bucket chain.
+    while ((pVolatileEntry = pVolatileEntry->m_pNextEntry) != nullptr)
```

Comment:
Does this need to be a volatile read? Otherwise, the C++ compiler optimizations are free to duplicate the read just like it was before the change.

Reply:
Yes, I would use volatile reads not only here, but in many places in this file.  It should better be a consistent effort though.